components/brave_wallet_ui/common/hooks/swap.ts

```diff
@@ -29,93 +31,172 @@ import { makeNetworkAsset } from '../../options/asset-options'
   const [exchangeRate, setExchangeRate] = React.useState('')
+  const [filteredAssetList, setFilteredAssetList] = React.useState<BraveWallet.BlockchainToken[]>([])
   const [fromAmount, setFromAmount] = React.useState('')
-  const [fromAsset, setFromAsset] = React.useState<BraveWallet.BlockchainToken | undefined>(swapAssetOptions[0])
+  const [fromAsset, setFromAsset] = React.useState<BraveWallet.BlockchainToken | undefined>(fromAssetProp)
+  const [isLoading, setIsLoading] = React.useState<boolean>(false)
+  const [isSupported, setIsSupported] = React.useState<boolean>(false)
   const [orderExpiration, setOrderExpiration] = React.useState<ExpirationPresetObjectType>(ExpirationPresetOptions[0])
   const [orderType, setOrderType] = React.useState<OrderTypes>('market')
+  const [selectedPreset, setSelectedPreset] = React.useState<AmountPresetTypes | undefined>(undefined)
   const [slippageTolerance, setSlippageTolerance] = React.useState<SlippagePresetObjectType>(SlippagePresetOptions[0])
-  const [customSlippageTolerance, setCustomSlippageTolerance] = React.useState<string>('')
-  const [toAmount, setToAmount] = React.useState('')
-  const [toAsset, setToAsset] = React.useState<BraveWallet.BlockchainToken | undefined>(swapAssetOptions[1])
-  const [filteredAssetList, setFilteredAssetList] = React.useState<BraveWallet.BlockchainToken[]>(swapAssetOptions)
+  const [swapError, setSwapError] = React.useState<SwapErrorResponse | undefined>(undefined)
+  const [swapQuote, setSwapQuote] = React.useState<BraveWallet.SwapResponse | undefined>(undefined)
   const [swapToOrFrom, setSwapToOrFrom] = React.useState<ToOrFromType>('from')
-  const [allowance, setAllowance] = React.useState<string | undefined>(undefined)
-  const [isLoading, setIsLoading] = React.useState<boolean>(false)
-  const [isSupported, setIsSupported] = React.useState<boolean>(false)
-
-  const nativeAsset = React.useMemo(
-    () => makeNetworkAsset(selectedNetwork),
-    [selectedNetwork]
-  )
-
-  React.useEffect(() => {
-    setFromAsset(swapAssetOptions[0])
-    setToAsset(swapAssetOptions[1])
-  }, [swapAssetOptions])
+  const [toAmount, setToAmount] = React.useState('')
+  const [toAsset, setToAsset] = React.useState<BraveWallet.BlockchainToken | undefined>(toAssetProp)
 
-  React.useEffect(() => {
-    isSwapSupported(selectedNetwork).then(
-      (supported) => setIsSupported(supported)
-    )
-  }, [selectedNetwork])
+  // custom hooks
+  const isMounted = useIsMounted()
+  const getBalance = useBalance(networkList)
+  const { getIsSwapSupported, getERC20Allowance } = useLib()
 
-  React.useEffect(() => {
-    if (!fromAsset) {
+  // Callbacks / methods
+  const fetchSwapQuote = React.useCallback(async (payload: SwapParamsPayloadType) => {
+    if (!isMounted) {
       return
     }
 
-    if (!fromAsset.isErc20) {
-      setAllowance(undefined)
-      return
-    }
+    const { swapService, ethTxManagerProxy } = getAPIProxy()
```

Comment:
This could be made available via a context, just like the async libs. For a later PR.